packages/fela/index.d.ts

```diff
@@ -1,13 +1,44 @@
-export interface Renderer {
-    clear(): void;
-    renderRule(rule: Function, props?: any): string;
-    renderKeyframe(keyframe: any, props?: any): string;
-    renderFont(family: string, files: any[], properties?: any): string;
-    renderStatic(style: string | any, selector?: string): string;
+declare module "fela" {
+
+  import React from 'react';
+
+  type TRuleProps = {};
+  type TRule = (props: TRuleProps) => IStyle; 
+  type TKeyFrame = TRule;
+  type TRendererCreator = (config?: IConfig) => IRenderer;
+  type TPlugin = (style: IStyle) => IStyle; //http://fela.js.org/docs/advanced/Plugins.html
+  type TEnhancer = (renderer: IRenderer) => IRenderer; //http://fela.js.org/docs/advanced/Enhancers.html
+
+  interface IRenderer {
+    renderRule(rule: TRule, props: TRuleProps): void;
+    renderKeyframe(keyFrame: TKeyFrame, props: TRuleProps): void;
+    renderFont(family: string, files: Array<string>, props: TRuleProps): void;
+    renderStatic(style: string, selector?: string): void;
     renderToString(): string;
-    subscribe(callback: Function): any;
+    subscribe(event: (msg: string) => void): { unsubscribe: () => void; }
```

Comment:
Sorry, I did not mention example on http://fela.js.org/docs/api/fela/Renderer.html#subscribelistener. I can see there: ```{ type: 'rule', style: 'font-size:12px', selector: 'a', media: '' }```. 

The only question is what to choose: ```declaration: 'color:red'``` or ```style: 'font-size:12px'``` ?

TS definition could be:
```
interface ISubscribeMessage {
    type: string;
    selector: string;
    style: string; //??? OR declaration:string; ???
    media: string;
}
...
subscribe(event: (msg: ISubscribeMessage) => void): { unsubscribe: () => void; }
```